visualization/app/codeCharta/ui/metricType/metricType.component.spec.ts

```diff
@@ -0,0 +1,207 @@
+import "./metricType.module"
+import { MetricTypeController } from "./metricType.component"
+import { getService, instantiateModule } from "../../../../mocks/ng.mockhelper"
+import { MetricService } from "../../state/metric.service"
+import { IRootScopeService } from "angular"
+import { SettingsService } from "../../state/settings.service"
+import { AttributeType, Settings } from "../../codeCharta.model"
+import { SETTINGS } from "../../util/dataMocks"
+import { CodeMapBuilding } from "../codeMap/rendering/codeMapBuilding"
+import { CodeMapMouseEventService } from "../codeMap/codeMap.mouseEvent.service"
+import { FileStateService } from "../../state/fileState.service"
+
+describe("MetricTypeController", () => {
+	let metricTypeController: MetricTypeController
+	let $rootScope: IRootScopeService
+	let metricService: MetricService
+	let settingsService: SettingsService
+
+	let settings: Settings
+
+	beforeEach(() => {
+		restartSystem()
+		rebuildController()
+		withMockedSettingsService()
+	})
+
+	function restartSystem() {
+		instantiateModule("app.codeCharta.ui.metricType")
+
+		$rootScope = getService<IRootScopeService>("$rootScope")
+		metricService = getService<MetricService>("metricService")
+
+		settings = JSON.parse(JSON.stringify(SETTINGS))
+	}
+
+	function rebuildController() {
+		metricTypeController = new MetricTypeController($rootScope, metricService)
+	}
+
+	function withMockedSettingsService() {
+		settingsService = metricTypeController["settingsService"] = jest.fn().mockReturnValue({
+			getSettings: jest.fn().mockReturnValue(settings)
+		})()
+	}
+
+	describe("constructor", () => {
+		beforeEach(() => {
+			SettingsService.subscribe = jest.fn()
+			CodeMapMouseEventService.subscribe = jest.fn()
+			FileStateService.subscribe = jest.fn()
+		})
+
+		it("should subscribe to SettingsService", () => {
+			rebuildController()
+
+			expect(SettingsService.subscribe).toHaveBeenCalledWith($rootScope, metricTypeController)
+		})
+
+		it("should subscribe to CodeMapMouseEventService", () => {
+			rebuildController()
+
+			expect(CodeMapMouseEventService.subscribe).toHaveBeenCalledWith($rootScope, metricTypeController)
+		})
+	})
+
+	describe("onSettingsChanged", () => {
+		it("should set the areaMetricType to absolute", () => {
+			metricTypeController.onSettingsChanged(settings, { dynamicSettings: { areaMetric: "rloc" } }, undefined)
+
+			expect(metricTypeController["_viewModel"].areaMetricType).toBe("absolute")
```

Comment:
Use of interface `AttributeType.absolute`?

Reply:
done